daemon/emer-machine-id-provider.c

```diff
@@ -292,18 +292,21 @@ read_one_machine_id (const gchar  *machine_id_path,
       return FALSE;
     }
 
+  if (machine_id_hex)
+    g_strlcpy (machine_id_hex, machine_id_sans_hyphens, sizeof(gchar)*FILE_LENGTH);
```

Comment:
In full knowledge that I originally suggested this array parameter approach, I think it would be better to accept a `char **` parameter and here use:

```suggestion
    *machine_id_hex = g_strchomp (g_steal_pointer (&machine_id_sans_hyphens));
```

on the basis that we have already allocated that string, and it saves having to think about whether the inbound array is big enough.

(For future reference, `sizeof(char) == 1` is always true because `sizeof` is defined to return the number of `char`s needed to store the given expression or type.)

Reply:
> In full knowledge that I originally suggested this array parameter approach, I think it would be better to accept a char ** parameter

So should I dynamically allocate the array right from the start? I think array allocated on function's stack(as of now) will not be able able passed upon without forced `(char **)`  pointer-conversion to the subsequent call functions

Reply:
Give the innermost function this signature:

```c
static gboolean
read_one_machine_id (const gchar  *machine_id_path,
                     gchar       **machine_id_hex,
                     uuid_t        id,
                     GError      **error)
```

And make the same change to all intervening function's signatures.

Change this line to:

```suggestion
    *machine_id_hex = g_strchomp (g_steal_pointer (&machine_id_sans_hyphens));
```

In `emer_daemon_get_tracking_id` define:

```c
   gchar *machine_id_hex = NULL;
```

Call:

```c
   if (emer_machine_id_provider_get_id (priv->machine_id_provider, &machine_id_hex, uuid)) ```

and then return it to the caller.

Reply:
addressed